lua/iswap.lua

```diff
@@ -262,58 +262,51 @@ function M.iswap_node(config, direction)
   vim.cmd([[silent! call repeat#set("\<Plug>ISwapNormal", -1)]])
 end
 
-
 -- TODO: refactor iswap() and iswap_with()
 -- swap current with one other node
-function M.iswap_with(direction, config)
+function M.imove_with(direction, config) M.iswap_with(direction, config, true) end
+function M.imove(direction, config) M.iswap(config, true) end
+
+function M.iswap_with(direction, config, imove)
```

Comment:
I'm not particularly fond of the change in function signature to accommodate `imove`. I think I would prefer a new function, but let me know if that makes less sense to you

Reply:
I could make functions `internal.iswap` and `internal.iswap_with` and then have the public API call those? I just noticed that everything except the last line where the swaps are performed wa fthe same between iswap and imove

Reply:
I saw a TODO comment for refactoring iswap and iswap_with to share more code so I thought it could be a separate PR/commit

Reply:
yeahh... i feel like i want to write some tests before I start attempting to merge all those `iswap_*` functions together though, for regressions. but writing tests is something i do not want to do ;)

Reply:
and yeah, the only difference between `IMove` and `ISwap` is basically:

for two nodes (`c: cursor node` and `a: selected node`):

IMove swaps each consecutive pair `c...a`

while ISwap just swaps `c` and `a`.

I think there should be something, maybe `swap` like `function iswap(...)` that's in `internal` and then all the `iswap_*` functions call that one but with custom callbacks or something..